PurchasesTests/Purchasing/ProductInfoTests.swift

```diff
@@ -4,126 +4,124 @@ import Nimble
+        expect(try productInfo.asDictionary()["intro_duration"] as? String) == introDuration
+        expect(try productInfo.asDictionary()["trial_duration"]).to(beNil())
     }
 
-    func testAsDictionaryDoesntAddIntroDurationIfDurationTypeNone() {
+    func testAsDictionaryDoesntAddIntroDurationIfDurationTypeNone() throws {
         let introDuration = "P3M"
         let productInfo: ProductInfo = .createMockProductInfo(introDuration: introDuration,
-                                                                introDurationType: .none)
-        expect(productInfo.asDictionary()["trial_duration"]).to(beNil())
-        expect(productInfo.asDictionary()["intro_duration"]).to(beNil())
+                                               introDurationType: .none)
+        expect(try productInfo.asDictionary()["trial_duration"]).to(beNil())
+        expect(try productInfo.asDictionary()["intro_duration"]).to(beNil())
     }
 
-    func testAsDictionaryConvertsIntroPriceCorrectly() {
+    func testAsDictionaryConvertsIntroPriceCorrectly() throws {
         let introPrice: NSDecimalNumber = 6.99
-        let productInfo: ProductInfo = .createMockProductInfo(introPrice: 6.99)
-        expect(productInfo.asDictionary()["introductory_price"] as? NSDecimalNumber) == introPrice
+        let productInfo: ProductInfo = .createMockProductInfo(introPrice: introPrice as Decimal)
+        expect(try productInfo.asDictionary()["introductory_price"] as? NSNumber) == introPrice
     }
 
     func testAsDictionaryConvertsSubscriptionGroupCorrectly() {
         let subscriptionGroup = "cool_group"
         let productInfo: ProductInfo = .createMockProductInfo(subscriptionGroup: subscriptionGroup)
-        expect(productInfo.asDictionary()["subscription_group_id"] as? String) == subscriptionGroup
+        expect(try productInfo.asDictionary()["subscription_group_id"] as? String) == subscriptionGroup
     }
 
     func testAsDictionaryConvertsDiscountsCorrectly() throws {
-        guard #available(iOS 12.2, macOS 10.14.4, tvOS 12.2, watchOS 6.2, *) else { return }
         let discount1 = PromotionalOffer(offerIdentifier: "offerid1",
-                                         price: 11,
+                                         price: 11.1,
                                          paymentMode: .payAsYouGo,
                                          subscriptionPeriod: .init(value: 1, unit: .month))
         
         let discount2 = PromotionalOffer(offerIdentifier: "offerid2",
-                                         price: 12,
+                                         price: 12.2,
                                          paymentMode: .payUpFront,
                                          subscriptionPeriod: .init(value: 5, unit: .week))
         
         let discount3 = PromotionalOffer(offerIdentifier: "offerid3",
-                                         price: 13,
+                                         price: 13.3,
```

Comment:
☝️ 3️⃣    👍 I like that you added the decimals 😄

Reply:
Never know when we're gonna lose precision 😅